### PR TITLE
Update stack_frame.dart to parse unexpected error format to null.

### LINKED ISSUE #131627: Failure to parse FireFox error
The bug blocks cl/551925539 that blocks https://github.com/flutter/flutter/issues/130354 that blocks https://github.com/flutter/devtools/issues/3951

Flutter seems to fail parsing Firefox of new version:

```
Assertion failed: "Expected wrapException@http://jfbxk5-20020a056670ad0500b00884f10d5997.prod.google.com:20563/mobile/flutter/samples/hello_flutter/app/integration_test/integration_tests_screendiff_test_testing_web_browsers_firefox_linux_v85_0_firefox-linux_dart2js_canvaskit_web_test.dart.dart2js.browser_test.dart.js:5864:43 to match RegExp/^(.+) (\\d+):(\\d+)\\s+(.+)$/."
** Note that stack traces are truncated. See [http://go/dart-chain-stack-traces](https://www.google.com/url?q=http://go/dart-chain-stack-traces&sa=D) for information on how to view the full stack trace **
g3-root/third_party/dart_lang/v2/sdk/lib/_internal/js_runtime/lib/js_helper.dart 1195:37   wrapException
g3-root/third_party/dart_lang/v2/sdk/lib/_internal/js_runtime/lib/js_helper.dart 2698:3    assertThrow
g3-root/third_party/dart_lang/v2/sdk/lib/internal/iterable.dart 393:20                     moveNext
g3-root/third_party/dart_lang/v2/sdk/lib/internal/iterable.dart 871:12                     moveNext
g3-root/third_party/dart_lang/v2/sdk/lib/_internal/js_runtime/lib/core_patch.dart 476:5    _of
g3-root/third_party/dart_lang/v2/sdk/lib/_internal/js_runtime/lib/core_patch.dart 457:39   of
g3-root/third_party/dart/flutter/lib/src/foundation/stack_frame.dart 73:5                  StackFrame.fromStackString
``` 

https://github.com/flutter/flutter/blob/d5ebf5395014302084bdf794f9cb19c4f14fc1f7/packages/flutter/lib/src/foundation/stack_frame.dart#L101

### PR DESCRIPTION
**Pull request by polina-c**
Fixes https://github.com/flutter/flutter/issues/131627

Originally this code sometimes was returning null and sometimes was failing, when stack frame is in unexpected format.

This PR updates for one of the code paths from failing to returning null.

### CODE REVIEW

File: packages/flutter/lib/src/foundation/stack_frame.dart [hunk 1]
```diff
@@ -82,28 +82,38 @@ class StackFrame {
         .toList();
   }
 
-  static StackFrame? _parseWebFrame(String line) {
+  /// Parses a single [StackFrame] from a line of a [StackTrace].
+  ///
+  /// Returns null if format is not as expected.
+  static StackFrame? _treyParseWebFrame(String line) {
```
Comment (natebosch) [inline packages/flutter/lib/src/foundation/stack_frame.dart:88]: If we are going to rename it,

```suggestion
  static StackFrame? _tryParseWebFrame(String line) {
```

I think we should either keep the existing name pattern, or also rename `_parseWebNonDebugFrame` for consistency within this file.

Comment (yjbanov) [inline packages/flutter/lib/src/foundation/stack_frame.dart:88]: nit: s/trey/try/

**Comment by polina-c** [comment]
Tested for Chrome: https://drive.google.com/file/d/18ggTn_nlF3dCxRN94X_ku35nGduNGAlS/view?usp=sharing

**Review by yjbanov — APPROVED** [review]
Thank you for checking! LGTM

**Review by annagrin — APPROVED** [review]
LGTM!

**Pull request merged**